rig-milvus/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+        }
+    }
+
+    /// Creates a Milvus semantic search request.
+    fn create_search_request(&self, data: Vec<f64>, limit: usize) -> SearchRequest {
+        SearchRequest {
+            collection_name: &self.collection_name,
+            db_name: &self.database_name,
+            data,
+            anns_field: "embedding",
+            limit,
+            output_fields: vec!["id", "distance", "document", "embeddedText"],
+        }
+    }
+
+    /// Creates a semantic search request, but only for IDs.
+    fn create_search_request_id_only(&self, data: Vec<f64>, limit: usize) -> SearchRequest {
+        SearchRequest {
+            collection_name: &self.collection_name,
+            db_name: &self.database_name,
+            data,
+            anns_field: "embedding",
+            limit,
+            output_fields: vec!["id", "distance"],
+        }
+    }
+
+    /// Insert vectors (with metadata) into your Milvus instance.
+    pub async fn insert_documents<Doc: Serialize + Embed + Send>(
+        &self,
+        documents: Vec<(Doc, OneOrMany<Embedding>)>,
+    ) -> Result<(), VectorStoreError> {
+        let url = format!(
+            "{base_url}/v2/vectordb/entities/insert",
+            base_url = self.base_url
+        );
+        let data = documents
+            .into_iter()
+            .flat_map(|(document, embeddings)| {
+                let json_document: serde_json::Value = serde_json::to_value(&document).unwrap();
+                let json_document_as_string = serde_json::to_string(&json_document).unwrap();
+
+                embeddings
+                    .into_iter()
+                    .map(|embedding| {
+                        let embedded_text = embedding.document;
+                        let embedding: Vec<f64> = embedding.vec;
+
+                        CreateRecord {
+                            document: json_document_as_string.clone(),
+                            embedded_text,
+                            embedding,
+                        }
+                    })
+                    .collect::<Vec<CreateRecord>>()
+            })
+            .collect::<Vec<CreateRecord>>();
+
+        let mut client = self.client.post(url);
+        if let Some(ref token) = self.token {
+            client = client.header("Authentication", format!("Bearer {token}"));
+        }
+
+        let insert_request = self.create_insert_request(data);
+
+        let body = serde_json::to_string(&insert_request).unwrap();
+
+        let res = client
+            .body(body)
+            .send()
+            .await
+            .map_err(|x| VectorStoreError::DatastoreError(x.to_string().into()))?;
+
+        if res.status() != StatusCode::OK {
+            let res = res.text().await.unwrap();
+
+            return Err(VectorStoreError::DatastoreError(res.to_string().into()));
+        }
+
+        Ok(())
+    }
+}
+
+impl<M: EmbeddingModel + std::marker::Sync + Send> VectorStoreIndex for MilvusVectorStore<M> {
```

Comment:
```suggestion
impl<M: EmbeddingModel + Sync + Send> VectorStoreIndex for MilvusVectorStore<M> {
```

i think we generally don't use full formed paths for `Send+Sync` anywhere else (unless this was copied from another implementation)

Reply:
Copied from the Qdrant impl - we might need to investigate and check whether or not the bounds are actually required in the Qdrant case in a later PR.